crates/core/src/delta_datafusion/mod.rs

```diff
@@ -505,22 +517,17 @@ impl<'a> DeltaScanBuilder<'a> {
         self
     }
 
-    /// Use the provided [SchemaRef] for the [DeltaScan]
-    pub fn with_schema(mut self, schema: SchemaRef) -> Self {
-        self.schema = Some(schema);
-        self
-    }
-
     pub async fn build(self) -> DeltaResult<DeltaScan> {
         let config = match self.config {
             Some(config) => config,
             None => DeltaScanConfigBuilder::new().build(self.snapshot)?,
         };
 
-        let schema = match self.schema {
-            Some(schema) => schema,
-            None => self.snapshot.arrow_schema()?,
-        };
+        let schema = config
+            .schema
+            .clone()
+            .unwrap_or(self.snapshot.arrow_schema()?);
```

Comment:
should be `unwrap_or_else` as to not eagerly call `self.snapshot.arrow_schema()` ?

Reply:
Oops, didn't realize unwrap_or always executes the evaluation call, will change to unwrap_or_else

Reply:
Changed it to match arm instead, unwrap_or_else and then error handling is weird unless I just unwrap the result of snapshot.arrow_schema() which could panic